docs/source/concepts/identity-and-package-management.rst

```diff
@@ -0,0 +1,170 @@
+
+Package Formats and Identifiers
+===============================
+
+Any code -- i.e., DAML templates -- to be uploaded must compiled down to the :ref:`DAML-LF <daml-lf>` language.
+The unit of packaging for DAML-LF is the :ref:`.dalf <dar-file-dalf-file>` file.
+Each ``.dalf`` file is uniquely identified by its **package identifier**, which is the hash of its contents.
+A :ref:`.dar <dar-file-dalf-file>` file is a simple archive containing multiple ``.dalf`` files, and has no identifier of its own.
+DAML ledgers support uploading only ``.dar`` files.
+
+Package Management API
+======================
+
+The package management API supports two methods:
+
+- ``UploadDarFile`` for uploading ``.dar`` files.
+  The ledger implementation is, however, free to reject any and all packages and return an error.
+  Furthermore, even if the method call succeeds, the ledger's vetting process might restrict the usability of the template.
+  For example, assume that Alice successfully uploads a ``.dar`` file to her participant containing a ``NewTemplate`` template.
+  It may happen that she can now issue commands that create ``NewTemplate`` instances with Bob as a stakeholder, but that all commands that create ``NewTemplate`` instances with Charlie as a stakeholder fail.
+
+- ``ListKnownPackages`` that lists the ``.dalf`` package vetted for usage at the participant node.
+  Like with the previous method, the usability of the listed templates depends on the ledger's vetting process.
+
+
+Package Vetting
+===============
+
+Using a DAML package entails running its DAML code.
+The DAML interpreter ensures that the DAML code cannot interact with the environment of the system on which it is executing.
+However, the operators of the ledger infrastructure nodes may still wish to review and vet any DAML code before allowing it to execute.
+One reason for this is that the DAML interpreter currently lacks a notion of reproducible resource limits.
+Thus, executing a DAML contract might result in high memory or CPU usage.
+Furthermore, security bugs in the DAML interpreter or JVM might enable malicious code to break out of the sandbox.
+
+Thus, DAML Ledgers generally allow some form of vetting a package before running its code on a node.
+Not all nodes in a DAML Ledger must vet all packages, as it is possible that some of them will not execute the code.
+For example, in :ref:`global state topologies <global-state-topologies>`, every :ref:`trust domain <trust-domain>` that controls how commits are appended to the shared ledger must execute DAML code.
+Thus, the operators of these trust domains will in general be allowed to vet the code before they execute it.
+The exact vetting mechanism is ledger-dependent.
+For example, in the :ref:`DAML Sandbox <sandbox-manual>`, the vetting is implicit: uploading a package through the Ledger API already vets the package, since its assumed that only the system administrator has access to these API facilities.
```

Comment:
```suggestion
For example, in the :ref:`DAML Sandbox <sandbox-manual>`, the vetting is implicit: uploading a package through the Ledger API already vets the package, since it's assumed that only the system administrator has access to these API facilities.
```